Classes/FrontendEnvironment/Tsfe.php

```diff
@@ -168,10 +168,10 @@ protected function initializeTsfe(int $pageId, int $language = 0, ?int $rootPage
             $backedUpBackendUser = $GLOBALS['BE_USER'] ?? null;
             try {
                 $tsfe->determineId($serverRequest);
-                $serverRequest->withAttribute('frontend.controller', $tsfe);
+                $serverRequest = $serverRequest->withAttribute('frontend.controller', $tsfe);
```

Comment:
Makes it sense to move this line above the `$tsfe->determineId($serverRequest);`?
Like?
```suggestion
                $serverRequest = $serverRequest->withAttribute('frontend.controller', $tsfe);
                $tsfe->determineId($serverRequest);
```